BUG: Goldy Bot still crashes when "raise_on_load_error" is set to false.

When `raise_on_load_error` is set to false goldy bot is meant to ignore the extension and continue loading other stuff but when the user forgets the `load()` function the command in the extension loads but the actual extension doesn't so it leads to this error below:

```python
[DEBUG] (Goldy Bot) - [Token] Searching current working directory for .env file if exists...
[DEBUG] (Goldy Bot) - [Token] Grabbing discord token...
[DEBUG] (Goldy Bot) - [Token] Grabbing database token...
[DEBUG] (Goldy Bot) - [Token] Done
[INFO] (Goldy Bot) - [Database] AsyncIOMotorClient Connected!
[DEBUG] (Goldy Bot) - [JSONConfig] Opening config file at './goldy.json'...
[DEBUG] (Goldy Bot) - [JSONConfig] Phrasing json in config to dict...
[DEBUG] (Goldy Bot) - [JSONConfig] Done!
[DEBUG] (Goldy Bot) - [Goldy] Nextcore shard manager connecting...
[DEBUG] (Goldy Bot) - [Goldy] Application data requested!
[DEBUG] (Goldy Bot) - [Database] [goldy_main] Found '{'_id': '863416692083916820'}' in 'guild_configs.'
[DEBUG] (Goldy Bot) - [Database] [goldy_main] Edited '{'_id': '863416692083916820'}' into '{'_id': '863416692083916820', 'code_name': 'test_server', 'prefix': '!', 'roles': {}, 'channels': {}, 'allowed_extensions': [], 'disallowed_extensions': [], 'hidden_extensions': []}.'
[INFO] (Goldy Bot) - [ExtensionLoader] Found extension folder at 'D:\test_goldybot\extensions'.
[DEBUG] (Goldy Bot) - [ExtensionLoader] Found the module 'test.py' in extensions.
[WARNING] (Goldy Bot) - [ExtensionLoader] Couldn't find extension path at 'D:\test_goldybot\env\Lib\site-packages\GoldyBot\internal_extensions'!
[DEBUG] (Goldy Bot) - [Command] [hello] Command initialized!
[DEBUG] (Goldy Bot) - [ExtensionLoader] Calling the 'load()' function in the extension at D:\test_goldybot\extensions\test.py...
[ERROR] (Goldy Bot) - [ExtensionLoader] We encountered an error while trying to load extension at 'D:\test_goldybot\extensions\test.py'! You most likely forgot the 'load()' function.
ERROR --> module 'D:\test_goldybot\extensions\test' has no attribute 'load'
Traceback (most recent call last):
  File "run.py", line 5, in <module>
    goldy.start() # Launch goldy bot.
  File "D:\test_goldybot\env\lib\site-packages\GoldyBot\goldy\__init__.py", line 96, in start
    self.async_loop.run_until_complete(
  File "C:\Users\jassi\AppData\Local\Programs\Python\Python38\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "D:\test_goldybot\env\lib\site-packages\GoldyBot\goldy\__init__.py", line 127, in __start_async
    await self.setup()
  File "D:\test_goldybot\env\lib\site-packages\GoldyBot\goldy\__init__.py", line 158, in setup
    await self.command_loader.load()
  File "D:\test_goldybot\env\lib\site-packages\GoldyBot\goldy\commands\loader.py", line 38, in load
    if command.extension.is_ignored == False:
  File "D:\test_goldybot\env\lib\site-packages\GoldyBot\goldy\commands\__init__.py", line 105, in extension
    return utils.cache_lookup(self.extension_name, extensions_cache)[1]
TypeError: 'NoneType' object is not subscriptable
```

Because the extension is not loaded when the command is loaded it cannot find the extension leading to this traceback.

--- a/GoldyBot/goldy/commands/__init__.py
+++ b/GoldyBot/goldy/commands/__init__.py
@@ -113,7 +113,7 @@ class Command():
     def extension(self) -> Extension | None:
         """Finds and returns the object of the command's extension. Returns None if command is not in any extension."""
         if self.in_extension:
-            return utils.cache_lookup(self.extension_name, extensions_cache)[1]
+            return (lambda x: x[1] if x is not None else None)(utils.cache_lookup(self.extension_name, extensions_cache))
         
         return None
 
@@ -266,13 +266,21 @@ class Command():
         return None
 
 
-    async def load(self) -> None:
+    async def load(self) -> bool:
         """Loads and creates the command."""
+        if self.in_extension:
+
+            if self.extension is None: # If the extension doesn't don't load this command.
+                self.logger.warn(
+                    f"Not loading command '{self.name}' because the extension '{self.extension_name}' is being ignored or has failed to load!"
+                )
+                return False
+
+            self.extension.commands.append(self)
+            self.logger.debug(f"Added to extension '{self.extension.code_name}'.")
+
         if self.allow_slash_cmd:
             await self.create_slash()
-
-        if self.extension is not None:
-            self.extension.add_command(self)
 
         self.__loaded = True
 
@@ -280,29 +288,21 @@ class Command():
             f"Command '{self.name}' has been loaded!"
         )
 
-        return None
+        return True
     
 
     async def unload(self) -> None:
-        """Unloads and removes the command."""
+        """Unloads and removes the command from cache."""
 
         if self.allow_slash_cmd:
             await self.remove_slash()
 
         self.__loaded = False
 
+        commands_cache.remove((self.name, self))
+
         self.logger.debug(
             f"Command '{self.name}' has been unloaded!"
         )
 
         return None
-    
-    async def delete(self) -> None:
-        """Completely deletes this command. Unloads it and removes it from cache."""
-        await self.unload()
-
-        commands_cache.remove((self.name, self))
-
-        self.logger.info(
-            f"Command '{self.name}' has been deleted!"
-        )

--- a/GoldyBot/goldy/commands/decorator.py
+++ b/GoldyBot/goldy/commands/decorator.py
@@ -24,10 +24,10 @@ def command(
     ⭐ Example:
     -------------
     This is how you create a command in GoldyBot::
-    
+
         @GoldyBot.command()
-        async def uwu(ctx):
-            await send(ctx, f'Hi, {ctx.author.mention}! UwU!')
+        async def hello(platter: GoldyBot.GoldenPlatter):
+            await platter.send_message("👋hello", reply=True)
     
     """
     def decorate(func):

--- a/GoldyBot/goldy/commands/loader.py
+++ b/GoldyBot/goldy/commands/loader.py
@@ -33,14 +33,9 @@ class CommandLoader():
 
         for command in commands:
 
-            if command.loaded == False:
+            if not command.loaded:
+                await command.load()
 
-                if command.extension.is_ignored == False:
-                    await command.load()
-                else:
-                    self.logger.debug(
-                        f"Not loading command '{command.name}' because the extension '{command.extension_name}' is being ignored!"
-                    )
             else:
                 self.logger.debug(
                     f"Not loading command '{command.name}' as it's already loaded."

--- a/GoldyBot/goldy/extensions/__init__.py
+++ b/GoldyBot/goldy/extensions/__init__.py
@@ -44,7 +44,6 @@ class Extension():
     def __init__(self):
         """Tells Goldy Bot to Load this class as an extension."""
         self.goldy = get_goldy_instance()
-        self.ignored_extensions_list = self.goldy.config.ignored_extensions
 
         self.logger = LoggerAdapter(
             LoggerAdapter(goldy_bot_logger, prefix = "Extensions"), 
@@ -52,15 +51,15 @@ class Extension():
         )
 
         # Cached commands list.
-        self.__commands:List[Command] = [
+        self.commands: List[Command] = [
 
         ]
 
         self.__loaded_path = os.path.realpath(self.__class__.__module__) + ".py"
 
-        self.__ignored = False
-        if self.code_name in self.ignored_extensions_list:
-            self.__ignored = True
+        if self.code_name.lower() in [extension.lower() for extension in self.goldy.config.ignored_extensions]:
+            self.logger.info(f"Not loading the extension '{self.code_name}' as it's ignored.")
+            return False
 
         # Adding to cache and loading commands.
         # ---------------------------------------        
@@ -77,28 +76,15 @@ class Extension():
     def loaded_path(self) -> str:
         "The path where this extension was loaded."
         return self.__loaded_path
-    
-    @property
-    def is_ignored(self):
-        """Returns if the extension is being ignored."""
-        return self.__ignored
 
-    def add_command(self, command:Command) -> None:
-        """Add this command to this extension."""
-        self.__commands.append(command)
-        self.logger.debug(f"Added the command '{command.name}' to {self.code_name}.")
-        return None
-    
-    def get_commands(self) -> List[Command]:
-        """Returns all the commands loaded with this extension."""
-        return self.__commands
-
-    async def delete(self) -> None:
+    async def unload(self) -> None:
         """Unloads and deletes itself from cache and all the commands with it."""
-        for command in self.get_commands():
-            await command.delete()
+        for command in self.commands:
+            await command.unload()
 
-        extensions_cache.remove((self.code_name, self))
+        extensions_cache.remove(
+            (self.code_name, self)
+        )
 
         self.logger.debug(f"Extension '{self.code_name}' removed!")
 

--- a/GoldyBot/goldy/extensions/extension_reloader.py
+++ b/GoldyBot/goldy/extensions/extension_reloader.py
@@ -39,9 +39,10 @@ class ExtensionReloader():
         self.logger.info(f"Reloading these extensions --> {[x.code_name for x in extensions]}")
 
         for extension in extensions:
-            # Delete all commands in extension.
-            await extension.delete()
+            # Unload all commands in extension.
+            await extension.unload()
 
+            # Get the full path the extension was loaded from so we can load it again with ExtensionLoader().
             if not extension.loaded_path in loaded_paths:
                 loaded_paths.append(extension.loaded_path)
 

